hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/shell/BulkDeleteCommand.java

```diff
@@ -88,56 +116,78 @@ protected LinkedList<PathData> expandArguments(LinkedList<String> args) throws I
         return pathData;
     }
 
+    /**
+     * Deletes the objects using the bulk delete api
+     * @param bulkDelete Bulkdelete object exposing the API
+     * @param paths list of paths to be deleted in the base path
+     * @throws IOException on error in execution of the delete command
+     */
     void deleteInBatches(BulkDelete bulkDelete, List<Path> paths) throws IOException {
         Batch<Path> batches = new Batch<>(paths, pageSize);
         while(batches.hasNext()) {
-            bulkDelete.bulkDelete(batches.next());
+            List<Map.Entry<Path, String>> result = bulkDelete.bulkDelete(batches.next());
```

Comment:
bulkDelete.bulkDelete() throws IllegalArgumentException as well.

Reply:
Added a catch clause around it and logged the error, should it fail completely here or not, if a single batch fails?

Reply:
Should fail here. no point trying other batches.